geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TileLayer.scala

```diff
@@ -239,4 +242,42 @@ object TileLayer {
       case NOCOMPRESSION => NoCompression
       case DEFLATECOMPRESSION => DeflateCompression
     }
+
+  def combineBands[K: ClassTag, L <: TileLayer[K]: ClassTag](
+    sc: SparkContext,
+    layers: ArrayList[L]
+  ): RDD[(K, MultibandTile)] = {
+    val scalaLayers = layers.asScala.toArray
+
+    val rdds: Array[RDD[(K, MultibandTile)]] =
+      scalaLayers.map { case (v: L) => v.rdd }
+
+    val arr = Array.ofDim[RDD[(K, (Int, MultibandTile))]](rdds.size)
+
+    for ((layer, index) <- rdds.zipWithIndex) {
+      arr(index) = layer.mapValues { (index, _) }
+    }
+
+    val unioned = sc.union(arr.toSeq)
+
+    val bands: RDD[(K, Map[Int, Vector[Tile]])] =
+      unioned.combineByKey(
+        (value: (Int, MultibandTile)) =>
+          Map(value._1 -> value._2.bands),
+        (bandMap: Map[Int, Vector[Tile]], value: (Int, MultibandTile)) =>
+          bandMap + (value._1 -> value._2.bands),
+        (m1: Map[Int, Vector[Tile]], m2: Map[Int, Vector[Tile]]) =>
+          m1 ++ m2
+      )
+
+    bands.mapValues { case (v: Map[Int, Vector[Tile]]) =>
+      MultibandTile(
+        v.toSeq
+          .sortWith(_._1 < _._1)
+          .toMap
```

Comment:
I believe turning the collection back to map may undo the sort and isn't actually required step here.

Reply:
Ah, I didn't know that. I'll fix it.